packages/vue-components/src/ScrollTopButton.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <i
+    id="scroll-top-button"
+    :class="[icon, getIconSize(), 'd-print-none', {'lighten': $data.lighten}]"
+    :style="iconStyle()"
+    aria-hidden="true"
+    @click="handleScrollTop()"
+  >
+  </i>
+</template>
+
+<script>
+export default {
+  name: 'ScrollTopButton.vue',
+  props: {
+    icon: {
+      default: 'fa fa-arrow-circle-up',
+      type: String,
+    },
+    iconSize: {
+      type: String,
+      default: 'lg',
+    },
+    iconColor: {
+      type: String,
+      default: null,
+    },
+    bottom: {
+      type: String,
+      default: null,
+    },
+    right: {
+      type: String,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      visible: false,
+      lighten: false,
```

Comment:
Minor nit, can we follow the CS2103 convention of having verbs in front of bool variables? e.g. `isVisible`